sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_client.py

```diff
@@ -353,8 +353,12 @@ def list_deleted_keys(self, **kwargs):
                 :dedent: 8
         """
         max_page_size = kwargs.get("max_page_size", None)
-        pages = self._client.get_deleted_keys(self._vault_url, maxresults=max_page_size, **kwargs)
-        return (DeletedKey._from_deleted_key_item(item) for item in pages)
+        return self._client.get_deleted_keys(
+            self._vault_url,
+            maxresults=max_page_size,
+            cls=lambda objs: [DeletedKey._from_deleted_key_item(x) for x in objs],
```

Comment:
I assume `objs` is a single page already received, so the comprehension doesn't entail additional I/O?

Reply:
correct, `objs` is a real list already in memory, from a single JSON